src/Campaigns/Routes/GetCampaignsListTable.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Give\Campaigns\Routes;
+
+use Give\API\RestRoute;
+use Give\Campaigns\ListTable\CampaignsListTable;
+use Give\Campaigns\Repositories\CampaignRepository;
+use Give\Framework\Database\DB;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class GetCampaignsListTable implements RestRoute
+{
+    /**
+     * @var string
+     */
+    protected $endpoint = 'campaigns/list-table';
+
+    /**
+     * @var WP_REST_Request
+     */
+    protected $request;
+
+    /**
+     * @var CampaignsListTable
+     */
+    protected $listTable;
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute(): void
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => [$this, 'permissionsCheck'],
+                ],
+                'args' => [
+                    'page' => [
+                        'type' => 'integer',
+                        'default' => 1,
+                        'minimum' => 1,
+                    ],
+                    'perPage' => [
+                        'type' => 'integer',
+                        'default' => 30,
+                        'minimum' => 1,
+                    ],
+                    'search' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortColumn' => [
+                        'type' => 'string',
+                        'default' => 'id',
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortDirection' => [
+                        'type' => 'string',
+                        'default' => 'asc',
+                        'enum' => ['asc', 'desc'],
+                    ],
+                    'locale' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'default' => get_locale(),
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function handleRequest(WP_REST_Request $request): WP_REST_Response
+    {
+        $this->request = $request;
+        $this->listTable = give(CampaignsListTable::class);
+
+        $campaigns = $this->getCampaigns();
+        $campaignsCount = $this->getTotalCampaignsCount();
+        $pageCount = (int)ceil($campaignsCount / $request->get_param('perPage'));
+
+        if ('model' === $this->request->get_param('return')) {
```

Comment:
Not sure why we have this.

Reply:
To be honest, I also don't know, but I believe it can be used somehow for our List Table component. I didn't check it deeper, but if you run a search in the codebase looking for `'model' === $this->request->get_param('return')` you'll see that it's used in all ListTables implemented.

Reply:
I'm unsure why this exists in other places, but it's wrong. PHP does not have a `__toArray` magic method to convert objects to an array. Outputting an array of objects like that will not do anything. So yeah, go ahead and remove this. 
Maybe I'm missing something so I will ask for clarification in Slack from others on why we have this in the first place.

Reply:
I agree with you, and thinking about what you said about `__toArray` I searched on the tests and I could see it being used there as you can check here: https://github.com/impress-org/givewp/blob/develop/tests/Unit/DonationForms/Repositories/DonationFormsRepositoryTest.php#L43

So seems that the only reason to have this is to make the unit tests easier. Not sure if is useful for anything else, let's check with the team.

Reply:
@alaca Thank you! Resolve in 5d1b45f7538c90e27376b6941762543c7e232a9b